solr/core/src/test/org/apache/solr/cloud/TestLocalFSCloudBackupRestore.java

```diff
@@ -24,12 +24,20 @@
  * such file-system would be exposed via local file-system API.
  */
 public class TestLocalFSCloudBackupRestore extends AbstractCloudBackupRestoreTestCase {
+  private static String backupLocation;
 
   @BeforeClass
   public static void setupClass() throws Exception {
     configureCluster(NUM_SHARDS)// nodes
         .addConfig("conf1", TEST_PATH().resolve("configsets").resolve("cloud-minimal").resolve("conf"))
         .configure();
+
+    boolean whitespacesInPath = random().nextBoolean();
+    if (whitespacesInPath) {
+      backupLocation = createTempDir("my backup").toFile().getAbsolutePath();
```

Comment:
I'd use `backupLocation = createTempDir(...).toAbsolutePath().toString();` to get rid of legacy `java.io.File`